Intermittent Pants service failure

```
Exception message: service failure for <pants.pantsd.service.scheduler_service.SchedulerService object at 0x112ed6048>, shutting down!
```
Started seeing this error some time after we upgraded to python3, and before we upgraded to `1.20.0.dev1` (so... May? ish? I can probably nail that down a bit.)

```sh
./pants test src/python/toolchain/satresolver::                                                                       

17:33:12 00:00 [main]
               See a report at: http://localhost:51055/run/pants_run_2019_08_22_17_33_12_748_d45583545b6f4bf4ae453f0e9415bf4b
17:33:12 00:00   [setup]
17:33:12 00:00     [parse]21:33:12 [WARN] /Users/tansypants/.cache/pants/setup/bootstrap-Darwin-x86_64/1.20.0.dev1_py37/lib/python3.7/site-packages/pants/option/options.py:460: DeprecationWarning: DEPRECATED: scope compile.zinc will be removed in version 1.20.0.dev2.
  Use scope compile.rsc instead (options: args, use_classpath_jars, incremental_caching)
  self._check_and_apply_deprecations(scope, values)


               Executing tasks in goals: bootstrap -> imports -> unpack-jars -> unpack-wheels -> deferred-sources -> native-compile -> link -> jvm-platform-validate -> gen -> resolve -> resources -> pyprep -> compile -> test21:33:13 [ERROR]
21:33:13 [ERROR] lost active connection to pantsd!
timestamp: 2019-08-22T17:33:13.168851
Exception caught: (pants.bin.remote_pants_runner.Terminated)
  File "/Users/tansypants/.cache/pants/setup/bootstrap-Darwin-x86_64/1.20.0.dev1_py37/bin/pants", line 10, in <module>
    sys.exit(main())
  File "/Users/tansypants/.cache/pants/setup/bootstrap-Darwin-x86_64/1.20.0.dev1_py37/lib/python3.7/site-packages/pants/bin/pants_loader.py", line 78, in main
    PantsLoader.run()
  File "/Users/tansypants/.cache/pants/setup/bootstrap-Darwin-x86_64/1.20.0.dev1_py37/lib/python3.7/site-packages/pants/bin/pants_loader.py", line 74, in run
    cls.load_and_execute(entrypoint)
  File "/Users/tansypants/.cache/pants/setup/bootstrap-Darwin-x86_64/1.20.0.dev1_py37/lib/python3.7/site-packages/pants/bin/pants_loader.py", line 67, in load_and_execute
    entrypoint_main()
  File "/Users/tansypants/.cache/pants/setup/bootstrap-Darwin-x86_64/1.20.0.dev1_py37/lib/python3.7/site-packages/pants/bin/pants_exe.py", line 32, in main
    PantsRunner(start_time=start_time).run()
  File "/Users/tansypants/.cache/pants/setup/bootstrap-Darwin-x86_64/1.20.0.dev1_py37/lib/python3.7/site-packages/pants/bin/pants_runner.py", line 67, in run
    return RemotePantsRunner(self._exiter, self._args, self._env, options_bootstrapper).run()
  File "/Users/tansypants/.cache/pants/setup/bootstrap-Darwin-x86_64/1.20.0.dev1_py37/lib/python3.7/site-packages/pants/bin/remote_pants_runner.py", line 194, in run
    self._run_pants_with_retry(pantsd_handle)
  File "/Users/tansypants/.cache/pants/setup/bootstrap-Darwin-x86_64/1.20.0.dev1_py37/lib/python3.7/site-packages/pants/bin/remote_pants_runner.py", line 137, in _run_pants_with_retry
    raise self._extract_remote_exception(pantsd_handle.pid, e).with_traceback(traceback)
  File "/Users/tansypants/.cache/pants/setup/bootstrap-Darwin-x86_64/1.20.0.dev1_py37/lib/python3.7/site-packages/pants/bin/remote_pants_runner.py", line 115, in _run_pants_with_retry
    return self._connect_and_execute(pantsd_handle)
  File "/Users/tansypants/.cache/pants/setup/bootstrap-Darwin-x86_64/1.20.0.dev1_py37/lib/python3.7/site-packages/pants/bin/remote_pants_runner.py", line 160, in _connect_and_execute
    result = client.execute(self.PANTS_COMMAND, *self._args, **modified_env)
  File "/Users/tansypants/.cache/pants/setup/bootstrap-Darwin-x86_64/1.20.0.dev1_py37/lib/python3.7/site-packages/pants/java/nailgun_client.py", line 364, in execute
    wrapped_exc=e,

Exception message: abruptly lost active connection to pantsd runner: NailgunError('Problem talking to nailgun server (address: 127.0.0.1:52826, remote_pid=34701, remote_pgrp=-34700): TruncatedHeaderError("Failed to read nailgun chunk header (TruncatedRead(\'Expected 5 bytes before socket shutdown, instead received 0\')).")', TruncatedHeaderError("Failed to read nailgun chunk header (TruncatedRead('Expected 5 bytes before socket shutdown, instead received 0'))."))
Remote exception:
timestamp: 2019-08-22T17:32:28.458611
process title: pantsd [/Users/tansypants/code/toolchain]
sys.argv: ['./pants', 'test', 'src/python/toolchain/satresolver::']
pid: 34701
Pants run failed with exception: Terminated with 1; exiting
timestamp: 2019-08-22T17:33:12.935813
process title: pantsd [/Users/tansypants/code/toolchain]
sys.argv: ['./pants', 'test', 'src/python/toolchain/satresolver::']
pid: 34701
Exception caught: (pants.pantsd.pants_daemon.RuntimeFailure)
  File "/Users/tansypants/.cache/pants/setup/bootstrap-Darwin-x86_64/1.20.0.dev1_py37/bin/pants", line 10, in <module>
    sys.exit(main())
  File "/Users/tansypants/.cache/pants/setup/bootstrap-Darwin-x86_64/1.20.0.dev1_py37/lib/python3.7/site-packages/pants/bin/pants_loader.py", line 78, in main
    PantsLoader.run()
  File "/Users/tansypants/.cache/pants/setup/bootstrap-Darwin-x86_64/1.20.0.dev1_py37/lib/python3.7/site-packages/pants/bin/pants_loader.py", line 74, in run
    cls.load_and_execute(entrypoint)
  File "/Users/tansypants/.cache/pants/setup/bootstrap-Darwin-x86_64/1.20.0.dev1_py37/lib/python3.7/site-packages/pants/bin/pants_loader.py", line 67, in load_and_execute
    entrypoint_main()
  File "/Users/tansypants/.cache/pants/setup/bootstrap-Darwin-x86_64/1.20.0.dev1_py37/lib/python3.7/site-packages/pants/pantsd/pants_daemon.py", line 510, in launch
    PantsDaemon.Factory.create(OptionsBootstrapper.create()).run_sync()
  File "/Users/tansypants/.cache/pants/setup/bootstrap-Darwin-x86_64/1.20.0.dev1_py37/lib/python3.7/site-packages/pants/pantsd/pants_daemon.py", line 443, in run_sync
    self._run_services(self._services)
  File "/Users/tansypants/.cache/pants/setup/bootstrap-Darwin-x86_64/1.20.0.dev1_py37/lib/python3.7/site-packages/pants/pantsd/pants_daemon.py", line 396, in _run_services
    raise PantsDaemon.RuntimeFailure('service failure for {}, shutting down!'.format(service))

Exception message: service failure for <pants.pantsd.service.scheduler_service.SchedulerService object at 0x112ed6048>, shutting down!
```

And again. This time it didn't even get far enough to yell at me about deprecated things.
```
./pants test src/python/toolchain/satresolver::                                                                                  
19:26:06 [ERROR]
19:26:06 [ERROR] lost active connection to pantsd!
timestamp: 2019-08-23T15:26:06.343224
Exception caught: (pants.bin.remote_pants_runner.Terminated)
  File "/Users/tansypants/.cache/pants/setup/bootstrap-Darwin-x86_64/1.20.0.dev1_py37/bin/pants", line 10, in <module>
    sys.exit(main())
  File "/Users/tansypants/.cache/pants/setup/bootstrap-Darwin-x86_64/1.20.0.dev1_py37/lib/python3.7/site-packages/pants/bin/pants_loader.py", line 78, in main
    PantsLoader.run()
  File "/Users/tansypants/.cache/pants/setup/bootstrap-Darwin-x86_64/1.20.0.dev1_py37/lib/python3.7/site-packages/pants/bin/pants_loader.py", line 74, in run
    cls.load_and_execute(entrypoint)
  File "/Users/tansypants/.cache/pants/setup/bootstrap-Darwin-x86_64/1.20.0.dev1_py37/lib/python3.7/site-packages/pants/bin/pants_loader.py", line 67, in load_and_execute
    entrypoint_main()
  File "/Users/tansypants/.cache/pants/setup/bootstrap-Darwin-x86_64/1.20.0.dev1_py37/lib/python3.7/site-packages/pants/bin/pants_exe.py", line 32, in main
    PantsRunner(start_time=start_time).run()
  File "/Users/tansypants/.cache/pants/setup/bootstrap-Darwin-x86_64/1.20.0.dev1_py37/lib/python3.7/site-packages/pants/bin/pants_runner.py", line 67, in run
    return RemotePantsRunner(self._exiter, self._args, self._env, options_bootstrapper).run()
  File "/Users/tansypants/.cache/pants/setup/bootstrap-Darwin-x86_64/1.20.0.dev1_py37/lib/python3.7/site-packages/pants/bin/remote_pants_runner.py", line 194, in run
    self._run_pants_with_retry(pantsd_handle)
  File "/Users/tansypants/.cache/pants/setup/bootstrap-Darwin-x86_64/1.20.0.dev1_py37/lib/python3.7/site-packages/pants/bin/remote_pants_runner.py", line 137, in _run_pants_with_retry
    raise self._extract_remote_exception(pantsd_handle.pid, e).with_traceback(traceback)
  File "/Users/tansypants/.cache/pants/setup/bootstrap-Darwin-x86_64/1.20.0.dev1_py37/lib/python3.7/site-packages/pants/bin/remote_pants_runner.py", line 115, in _run_pants_with_retry
    return self._connect_and_execute(pantsd_handle)
  File "/Users/tansypants/.cache/pants/setup/bootstrap-Darwin-x86_64/1.20.0.dev1_py37/lib/python3.7/site-packages/pants/bin/remote_pants_runner.py", line 160, in _connect_and_execute
    result = client.execute(self.PANTS_COMMAND, *self._args, **modified_env)
  File "/Users/tansypants/.cache/pants/setup/bootstrap-Darwin-x86_64/1.20.0.dev1_py37/lib/python3.7/site-packages/pants/java/nailgun_client.py", line 364, in execute
    wrapped_exc=e,

Exception message: abruptly lost active connection to pantsd runner: NailgunError('Problem talking to nailgun server (address: 127.0.0.1:50045, remote_pid=4010, remote_pgrp=-4009): TruncatedHeaderError("Failed to read nailgun chunk header (TruncatedRead(\'Expected 5 bytes before socket shutdown, instead received 0\')).")', TruncatedHeaderError("Failed to read nailgun chunk header (TruncatedRead('Expected 5 bytes before socket shutdown, instead received 0'))."))
Remote exception:
timestamp: 2019-08-23T15:25:33.861203
process title: pantsd [/Users/tansypants/code/toolchain]
sys.argv: ['./pants', 'test', 'src/python/toolchain/satresolver::']
pid: 4010
Pants run failed with exception: Terminated with 1; exiting
timestamp: 2019-08-23T15:26:06.122568
process title: pantsd [/Users/tansypants/code/toolchain]
sys.argv: ['./pants', 'test', 'src/python/toolchain/satresolver::']
pid: 4010
Exception caught: (pants.pantsd.pants_daemon.RuntimeFailure)
  File "/Users/tansypants/.cache/pants/setup/bootstrap-Darwin-x86_64/1.20.0.dev1_py37/bin/pants", line 10, in <module>
    sys.exit(main())
  File "/Users/tansypants/.cache/pants/setup/bootstrap-Darwin-x86_64/1.20.0.dev1_py37/lib/python3.7/site-packages/pants/bin/pants_loader.py", line 78, in main
    PantsLoader.run()
  File "/Users/tansypants/.cache/pants/setup/bootstrap-Darwin-x86_64/1.20.0.dev1_py37/lib/python3.7/site-packages/pants/bin/pants_loader.py", line 74, in run
    cls.load_and_execute(entrypoint)
  File "/Users/tansypants/.cache/pants/setup/bootstrap-Darwin-x86_64/1.20.0.dev1_py37/lib/python3.7/site-packages/pants/bin/pants_loader.py", line 67, in load_and_execute
    entrypoint_main()
  File "/Users/tansypants/.cache/pants/setup/bootstrap-Darwin-x86_64/1.20.0.dev1_py37/lib/python3.7/site-packages/pants/pantsd/pants_daemon.py", line 510, in launch
    PantsDaemon.Factory.create(OptionsBootstrapper.create()).run_sync()
  File "/Users/tansypants/.cache/pants/setup/bootstrap-Darwin-x86_64/1.20.0.dev1_py37/lib/python3.7/site-packages/pants/pantsd/pants_daemon.py", line 443, in run_sync
    self._run_services(self._services)
  File "/Users/tansypants/.cache/pants/setup/bootstrap-Darwin-x86_64/1.20.0.dev1_py37/lib/python3.7/site-packages/pants/pantsd/pants_daemon.py", line 396, in _run_services
    raise PantsDaemon.RuntimeFailure('service failure for {}, shutting down!'.format(service))

Exception message: service failure for <pants.pantsd.service.scheduler_service.SchedulerService object at 0x110d08a20>, shutting down!
````

Log:
```
19:25:28 [INFO] starting service <pants.pantsd.service.store_gc_service.StoreGCService object at 0x110d08710>
19:25:28 [INFO] handling pailgun request: `./pants test src/python/toolchain/satresolver::`
19:25:29 [INFO] confirmed watchman subscription: {'clock': 'c:1566583299:1254:1:1734', 'subscribe': 'all_files', 'version': '4.9.1'}
19:25:29 [INFO] enqueuing 44843 changes for subscription all_files
19:25:29 [INFO] confirmed watchman subscription: {'clock': 'c:1566583299:1254:1:1736', 'subscribe': 'pantsd_pid', 'version': '4.9.1'}
19:25:29 [INFO] enqueuing 1 changes for subscription pantsd_pid
19:25:33 [INFO] pailgun request completed: `./pants test src/python/toolchain/satresolver::`
19:26:03 [INFO] enqueuing 1 changes for subscription all_files
19:26:03 [WARN] invalidation: cleared 2 and dirtied 21 nodes for: {"src/python/toolchain/satresolver/pypi/python_graph_test.py", "src/python/toolchain/satresolver/pypi"}
19:26:03 [ERROR] saw file events covered by invalidation globs [('pants.ini', 
```
Enormous list of files truncated. I hope they aren't actually useful. 
```
 'src/python/toolchain/workflow/worker_thread.py')], terminating the daemon.
19:26:03 [INFO] enqueuing 1 changes for subscription all_files
19:26:05 [INFO] handling pailgun request: `./pants test src/python/toolchain/satresolver::`
19:26:05 [INFO] terminating pantsd service: <pants.pantsd.service.fs_event_service.FSEventService object at 0x110d08ba8>
19:26:06 [INFO] terminating pantsd service: <pants.pantsd.service.scheduler_service.SchedulerService object at 0x110d08a20>
19:26:06 [INFO] terminating pantsd service: <pants.pantsd.service.pailgun_service.PailgunService object at 0x1106c4518>
19:26:06 [INFO] pailgun service on port 50045 shutting down
19:26:06 [INFO] terminating pantsd service: <pants.pantsd.service.store_gc_service.StoreGCService object at 0x110d08710>
19:26:06 [INFO] terminating pantsd

Not sure if this one belongs here, but it seems it may be related?

```
./pants test src/python/toolchain/satresolver::                                                                                
22:35:38 [ERROR]
22:35:38 [ERROR] lost active connection to pantsd!
timestamp: 2019-08-23T18:35:38.839124
Exception caught: (pants.bin.remote_pants_runner.Terminated)
  File "/Users/tansypants/.cache/pants/setup/bootstrap-Darwin-x86_64/1.20.0.dev1_py37/bin/pants", line 10, in <module>
    sys.exit(main())
  File "/Users/tansypants/.cache/pants/setup/bootstrap-Darwin-x86_64/1.20.0.dev1_py37/lib/python3.7/site-packages/pants/bin/pants_loader.py", line 78, in main
    PantsLoader.run()
  File "/Users/tansypants/.cache/pants/setup/bootstrap-Darwin-x86_64/1.20.0.dev1_py37/lib/python3.7/site-packages/pants/bin/pants_loader.py", line 74, in run
    cls.load_and_execute(entrypoint)
  File "/Users/tansypants/.cache/pants/setup/bootstrap-Darwin-x86_64/1.20.0.dev1_py37/lib/python3.7/site-packages/pants/bin/pants_loader.py", line 67, in load_and_execute
    entrypoint_main()
  File "/Users/tansypants/.cache/pants/setup/bootstrap-Darwin-x86_64/1.20.0.dev1_py37/lib/python3.7/site-packages/pants/bin/pants_exe.py", line 32, in main
    PantsRunner(start_time=start_time).run()
  File "/Users/tansypants/.cache/pants/setup/bootstrap-Darwin-x86_64/1.20.0.dev1_py37/lib/python3.7/site-packages/pants/bin/pants_runner.py", line 67, in run
    return RemotePantsRunner(self._exiter, self._args, self._env, options_bootstrapper).run()
  File "/Users/tansypants/.cache/pants/setup/bootstrap-Darwin-x86_64/1.20.0.dev1_py37/lib/python3.7/site-packages/pants/bin/remote_pants_runner.py", line 194, in run
    self._run_pants_with_retry(pantsd_handle)
  File "/Users/tansypants/.cache/pants/setup/bootstrap-Darwin-x86_64/1.20.0.dev1_py37/lib/python3.7/site-packages/pants/bin/remote_pants_runner.py", line 137, in _run_pants_with_retry
    raise self._extract_remote_exception(pantsd_handle.pid, e).with_traceback(traceback)
  File "/Users/tansypants/.cache/pants/setup/bootstrap-Darwin-x86_64/1.20.0.dev1_py37/lib/python3.7/site-packages/pants/bin/remote_pants_runner.py", line 115, in _run_pants_with_retry
    return self._connect_and_execute(pantsd_handle)
  File "/Users/tansypants/.cache/pants/setup/bootstrap-Darwin-x86_64/1.20.0.dev1_py37/lib/python3.7/site-packages/pants/bin/remote_pants_runner.py", line 160, in _connect_and_execute
    result = client.execute(self.PANTS_COMMAND, *self._args, **modified_env)
  File "/Users/tansypants/.cache/pants/setup/bootstrap-Darwin-x86_64/1.20.0.dev1_py37/lib/python3.7/site-packages/pants/java/nailgun_client.py", line 364, in execute
    wrapped_exc=e,

Exception message: abruptly lost active connection to pantsd runner: NailgunError("Problem talking to nailgun server (address: 127.0.0.1:51369, remote_pid=<remote PID chunk not yet received!>, remote_pgrp=<remote PGRP chunk not yet received!>): ConnectionResetError(54, 'Connection reset by peer')", ConnectionResetError(54, 'Connection reset by peer'))
Remote exception:
timestamp: 2019-08-23T18:35:27.618169
process title: pantsd [/Users/tansypants/code/toolchain]
sys.argv: ['./pants', 'test', 'src/python/toolchain/satresolver::']
pid: 14034
Pants run failed with exception: Terminated with 1; exiting
```

Log:

```
22:34:36 [INFO] setting up service <pants.pantsd.service.fs_event_service.FSEventService object at 0x10b8bff98>
22:34:36 [INFO] setting up service <pants.pantsd.service.scheduler_service.SchedulerService object at 0x10b8bfe10>
22:34:37 [INFO] watching invalidating files: ('pants.ini', 
22:34:37 [INFO] setting up service <pants.pantsd.service.pailgun_service.PailgunService object at 0x10b8bfa58>
22:34:37 [INFO] setting up service <pants.pantsd.service.store_gc_service.StoreGCService object at 0x10b8bf940>
22:34:37 [INFO] starting service <pants.pantsd.service.fs_event_service.FSEventService object at 0x10b8bff98>
22:34:37 [INFO] starting service <pants.pantsd.service.scheduler_service.SchedulerService object at 0x10b8bfe10>
22:34:37 [INFO] starting service <pants.pantsd.service.pailgun_service.PailgunService object at 0x10b8bfa58>
22:34:37 [INFO] starting pailgun server on port 51352
22:34:37 [INFO] starting service <pants.pantsd.service.store_gc_service.StoreGCService object at 0x10b8bf940>
22:34:37 [INFO] handling pailgun request: `./pants test src/python/toolchain/satresolver::`
22:34:37 [INFO] confirmed watchman subscription: {'clock': 'c:1566583299:1254:1:7292', 'subscribe': 'all_files', 'version': '4.9.1'}
22:34:37 [INFO] enqueuing 44842 changes for subscription all_files
22:34:38 [INFO] confirmed watchman subscription: {'clock': 'c:1566583299:1254:1:7294', 'subscribe': 'pantsd_pid', 'version': '4.9.1'}
22:34:38 [INFO] enqueuing 1 changes for subscription pantsd_pid
22:34:43 [INFO] pailgun request completed: `./pants test src/python/toolchain/satresolver::`
22:35:16 [INFO] enqueuing 1 changes for subscription all_files
22:35:16 [WARN] invalidation: cleared 2 and dirtied 28 nodes for: {"src/python/toolchain/satresolver/pypi", "src/python/toolchain/satresolver/pypi/python_graph_test.py"}
22:35:17 [ERROR] saw file events covered by invalidation globs [('pants.ini', 
```
truncated.

```
'src/python/toolchain/workflow/worker.py', 'src/python/toolchain/workflow/worker_thread.py')], terminating the daemon.
22:35:18 [INFO] terminating pantsd service: <pants.pantsd.service.fs_event_service.FSEventService object at 0x10b8bff98>
22:35:18 [INFO] terminating pantsd service: <pants.pantsd.service.scheduler_service.SchedulerService object at 0x10b8bfe10>
22:35:18 [INFO] terminating pantsd service: <pants.pantsd.service.pailgun_service.PailgunService object at 0x10b8bfa58>
22:35:18 [INFO] pailgun service on port 51352 shutting down
22:35:18 [INFO] terminating pantsd service: <pants.pantsd.service.store_gc_service.StoreGCService object at 0x10b8bf940>
22:35:18 [INFO] terminating pantsd
22:35:18 [WARN] Encountered error when trying to log this message: b'timestamp: 2019-08-23T18:35:18.225449\nException caught: (pants.pantsd.pants_daemon.RuntimeFailure)\n  File "/Users/tansypants/.cache/pants/setup/bootstrap-Darwin-x86_64/1.20.0.dev1_py37/bin/pants", line 10, in <module>\n    sys.exit(main())\n  File "/Users/tansypants/.cache/pants/setup/bootstrap-Darwin-x86_64/1.20.0.dev1_py37/lib/python3.7/site-packages/pants/bin/pants_loader.py", line 78, in main\n    PantsLoader.run()\n  File "/Users/tansypants/.cache/pants/setup/bootstrap-Darwin-x86_64/1.20.0.dev1_py37/lib/python3.7/site-packages/pants/bin/pants_loader.py", line 74, in run\n    cls.load_and_execute(entrypoint)\n  File "/Users/tansypants/.cache/pants/setup/bootstrap-Darwin-x86_64/1.20.0.dev1_py37/lib/python3.7/site-packages/pants/bin/pants_loader.py", line 67, in load_and_execute\n    entrypoint_main()\n  File "/Users/tansypants/.cache/pants/setup/bootstrap-Darwin-x86_64/1.20.0.dev1_py37/lib/python3.7/site-packages/pants/pantsd/pants_daemon.py", line 510, in launch\n    PantsDaemon.Factory.create(OptionsBootstrapper.create()).run_sync()\n  File "/Users/tansypants/.cache/pants/setup/bootstrap-Darwin-x86_64/1.20.0.dev1_py37/lib/python3.7/site-packages/pants/pantsd/pants_daemon.py", line 443, in run_sync\n    self._run_services(self._services)\n  File "/Users/tansypants/.cache/pants/setup/bootstrap-Darwin-x86_64/1.20.0.dev1_py37/lib/python3.7/site-packages/pants/pantsd/pants_daemon.py", line 396, in _run_services\n    raise PantsDaemon.RuntimeFailure(\'service failure for {}, shutting down!\'.format(service))\n\nException message: service failure for <pants.pantsd.service.scheduler_service.SchedulerService object at 0x10b8bfe10>, shutting down!\n\n',
 exception: write to closed file
 out: <_io.BufferedWriter name=2>
22:35:21 [INFO] setting up service <pants.pantsd.service.fs_event_service.FSEventService object at 0x106c92208>
22:35:21 [INFO] setting up service <pants.pantsd.service.scheduler_service.SchedulerService object at 0x106c920b8>
22:35:21 [INFO] watching invalidating files: ('pants.ini', 
```
truncated. (These all appear to be the complete list of every file in our repo.)
```
22:35:21 [INFO] setting up service <pants.pantsd.service.pailgun_service.PailgunService object at 0x106a32c88>
22:35:21 [INFO] setting up service <pants.pantsd.service.store_gc_service.StoreGCService object at 0x106a32b70>
22:35:21 [INFO] starting service <pants.pantsd.service.fs_event_service.FSEventService object at 0x106c92208>
22:35:21 [INFO] starting service <pants.pantsd.service.scheduler_service.SchedulerService object at 0x106c920b8>
22:35:21 [INFO] starting service <pants.pantsd.service.pailgun_service.PailgunService object at 0x106a32c88>
22:35:21 [INFO] starting pailgun server on port 51369
22:35:21 [INFO] starting service <pants.pantsd.service.store_gc_service.StoreGCService object at 0x106a32b70>
22:35:21 [INFO] handling pailgun request: `./pants test src/python/toolchain/satresolver::`
22:35:22 [INFO] confirmed watchman subscription: {'clock': 'c:1566583299:1254:1:7408', 'subscribe': 'all_files', 'version': '4.9.1'}
22:35:22 [INFO] enqueuing 44842 changes for subscription all_files
22:35:22 [INFO] confirmed watchman subscription: {'clock': 'c:1566583299:1254:1:7410', 'subscribe': 'pantsd_pid', 'version': '4.9.1'}
22:35:22 [INFO] enqueuing 1 changes for subscription pantsd_pid
22:35:27 [INFO] pailgun request completed: `./pants test src/python/toolchain/satresolver::`
22:35:36 [INFO] enqueuing 1 changes for subscription all_files
22:35:36 [WARN] invalidation: cleared 2 and dirtied 28 nodes for: {"src/python/toolchain/satresolver/pypi", "src/python/toolchain/satresolver/pypi/python_graph_test.py"}
22:35:36 [ERROR] saw file events covered by invalidation globs [('pants.ini', 
```
truncated
```
22:35:38 [INFO] terminating pantsd service: <pants.pantsd.service.fs_event_service.FSEventService object at 0x106c92208>
22:35:38 [INFO] terminating pantsd service: <pants.pantsd.service.scheduler_service.SchedulerService object at 0x106c920b8>
22:35:38 [INFO] terminating pantsd service: <pants.pantsd.service.pailgun_service.PailgunService object at 0x106a32c88>
22:35:39 [INFO] terminating pantsd service: <pants.pantsd.service.store_gc_service.StoreGCService object at 0x106a32b70>
22:35:39 [INFO] terminating pantsd
22:35:39 [INFO] pailgun service on port 51369 shutting down
22:35:39 [WARN] Encountered error when trying to log this message: b'timestamp: 2019-08-23T18:35:39.843671\nException caught: (pants.pantsd.pants_daemon.RuntimeFailure)\n  File "/Users/tansypants/.cache/pants/setup/bootstrap-Darwin-x86_64/1.20.0.dev1_py37/bin/pants", line 10, in <module>\n    sys.exit(main())\n  File "/Users/tansypants/.cache/pants/setup/bootstrap-Darwin-x86_64/1.20.0.dev1_py37/lib/python3.7/site-packages/pants/bin/pants_loader.py", line 78, in main\n    PantsLoader.run()\n  File "/Users/tansypants/.cache/pants/setup/bootstrap-Darwin-x86_64/1.20.0.dev1_py37/lib/python3.7/site-packages/pants/bin/pants_loader.py", line 74, in run\n    cls.load_and_execute(entrypoint)\n  File "/Users/tansypants/.cache/pants/setup/bootstrap-Darwin-x86_64/1.20.0.dev1_py37/lib/python3.7/site-packages/pants/bin/pants_loader.py", line 67, in load_and_execute\n    entrypoint_main()\n  File "/Users/tansypants/.cache/pants/setup/bootstrap-Darwin-x86_64/1.20.0.dev1_py37/lib/python3.7/site-packages/pants/pantsd/pants_daemon.py", line 510, in launch\n    PantsDaemon.Factory.create(OptionsBootstrapper.create()).run_sync()\n  File "/Users/tansypants/.cache/pants/setup/bootstrap-Darwin-x86_64/1.20.0.dev1_py37/lib/python3.7/site-packages/pants/pantsd/pants_daemon.py", line 443, in run_sync\n    self._run_services(self._services)\n  File "/Users/tansypants/.cache/pants/setup/bootstrap-Darwin-x86_64/1.20.0.dev1_py37/lib/python3.7/site-packages/pants/pantsd/pants_daemon.py", line 396, in _run_services\n    raise PantsDaemon.RuntimeFailure(\'service failure for {}, shutting down!\'.format(service))\n\nException message: service failure for <pants.pantsd.service.scheduler_service.SchedulerService object at 0x106c920b8>, shutting down!\n\n',
 exception: write to closed file
 out: <_io.BufferedWriter name=2>
```

--- a/src/python/pants/engine/internals/native.py
+++ b/src/python/pants/engine/internals/native.py
@@ -1003,7 +1003,6 @@ class Native(metaclass=SingletonMetaclass):
             execution_options.process_execution_use_local_cache,
             self.context.utf8_dict(execution_options.remote_execution_headers),
             execution_options.process_execution_local_enable_nailgun,
-            execution_options.experimental_fs_watcher,
         )
         if scheduler_result.is_throw:
             value = self.context.from_value(scheduler_result.throw_handle)

--- a/src/python/pants/option/global_options.py
+++ b/src/python/pants/option/global_options.py
@@ -662,9 +662,11 @@ class GlobalOptions(Subsystem):
             "--watchman-enable",
             type=bool,
             advanced=True,
-            default=True,
+            default=False,
+            removal_version="1.29.0.dev2",
+            removal_hint="The native watcher is now sufficient to monitor for filesystem changes.",
             help="Use the watchman daemon filesystem event watcher to watch for changes "
-            "in the buildroot. Disable this to rely solely on the experimental pants engine filesystem watcher.",
+            "in the buildroot in addition to the built in watcher.",
         )
         register(
             "--watchman-version", advanced=True, default="4.9.0-pants1", help="Watchman version."

--- a/src/python/pants/option/global_options.py
+++ b/src/python/pants/option/global_options.py
@@ -102,7 +102,6 @@ class ExecutionOptions:
     remote_execution_extra_platform_properties: Any
     remote_execution_headers: Any
     process_execution_local_enable_nailgun: bool
-    experimental_fs_watcher: bool
 
     @classmethod
     def from_bootstrap_options(cls, bootstrap_options):
@@ -128,7 +127,6 @@ class ExecutionOptions:
             remote_execution_extra_platform_properties=bootstrap_options.remote_execution_extra_platform_properties,
             remote_execution_headers=bootstrap_options.remote_execution_headers,
             process_execution_local_enable_nailgun=bootstrap_options.process_execution_local_enable_nailgun,
-            experimental_fs_watcher=bootstrap_options.experimental_fs_watcher,
         )
 
 
@@ -154,7 +152,6 @@ DEFAULT_EXECUTION_OPTIONS = ExecutionOptions(
     remote_execution_extra_platform_properties=[],
     remote_execution_headers={},
     process_execution_local_enable_nailgun=False,
-    experimental_fs_watcher=True,
 )
 
 
@@ -903,6 +900,8 @@ class GlobalOptions(Subsystem):
             type=bool,
             default=True,
             advanced=True,
+            removal_version="1.29.0.dev2",
+            removal_hint="Enabled by default: flag is disabled.",
             help="Whether to use the engine filesystem watcher which registers the workspace"
             " for kernel file change events",
         )
@@ -1063,6 +1062,8 @@ class GlobalOptions(Subsystem):
         Raises pants.option.errors.OptionsError on validation failure.
         """
         if opts.get("loop") and not opts.enable_pantsd:
+            # TODO: This remains the case today because there are two spots that
+            # call `run_goal_rules`: fixing in a followup.
             raise OptionsError(
                 "The `--loop` option requires `--enable-pantsd`, in order to watch files."
             )

--- a/src/python/pants/option/global_options.py
+++ b/src/python/pants/option/global_options.py
@@ -663,7 +663,7 @@ class GlobalOptions(Subsystem):
             type=bool,
             advanced=True,
             default=False,
-            removal_version="1.29.0.dev2",
+            removal_version="1.30.0.dev0",
             removal_hint="The native watcher is now sufficient to monitor for filesystem changes.",
             help="Use the watchman daemon filesystem event watcher to watch for changes "
             "in the buildroot in addition to the built in watcher.",
@@ -902,7 +902,7 @@ class GlobalOptions(Subsystem):
             type=bool,
             default=True,
             advanced=True,
-            removal_version="1.29.0.dev2",
+            removal_version="1.30.0.dev0",
             removal_hint="Enabled by default: flag is disabled.",
             help="Whether to use the engine filesystem watcher which registers the workspace"
             " for kernel file change events",

--- a/src/rust/engine/engine_cffi/src/lib.rs
+++ b/src/rust/engine/engine_cffi/src/lib.rs
@@ -240,7 +240,6 @@ pub extern "C" fn scheduler_create(
   process_execution_use_local_cache: bool,
   remote_execution_headers_buf: BufferBuffer,
   process_execution_local_enable_nailgun: bool,
-  experimental_fs_watcher: bool,
 ) -> RawResult {
   match make_core(
     tasks_ptr,
@@ -271,7 +270,6 @@ pub extern "C" fn scheduler_create(
     process_execution_use_local_cache,
     remote_execution_headers_buf,
     process_execution_local_enable_nailgun,
-    experimental_fs_watcher,
   ) {
     Ok(core) => RawResult {
       is_throw: false,
@@ -315,7 +313,6 @@ fn make_core(
   process_execution_use_local_cache: bool,
   remote_execution_headers_buf: BufferBuffer,
   process_execution_local_enable_nailgun: bool,
-  experimental_fs_watcher: bool,
 ) -> Result<Core, String> {
   let root_type_ids = root_type_ids.to_vec();
   let ignore_patterns = ignore_patterns_buf
@@ -427,7 +424,6 @@ fn make_core(
     process_execution_use_local_cache,
     remote_execution_headers,
     process_execution_local_enable_nailgun,
-    experimental_fs_watcher,
   )
 }
 

--- a/src/rust/engine/src/context.rs
+++ b/src/rust/engine/src/context.rs
@@ -91,7 +91,6 @@ impl Core {
     process_execution_use_local_cache: bool,
     remote_execution_headers: BTreeMap<String, String>,
     process_execution_local_enable_nailgun: bool,
-    experimental_fs_watcher: bool,
   ) -> Result<Core, String> {
     // Randomize CAS address order to avoid thundering herds from common config.
     let mut remote_store_servers = remote_store_servers;
@@ -266,12 +265,7 @@ impl Core {
       GitignoreStyleExcludes::create_with_gitignore_file(ignore_patterns, gitignore_file)
         .map_err(|e| format!("Could not parse build ignore patterns: {:?}", e))?;
 
-    let watcher = InvalidationWatcher::new(
-      executor.clone(),
-      build_root.clone(),
-      ignorer.clone(),
-      experimental_fs_watcher,
-    )?;
+    let watcher = InvalidationWatcher::new(executor.clone(), build_root.clone(), ignorer.clone())?;
     watcher.start(&graph);
 
     Ok(Core {

--- a/src/rust/engine/watch/src/lib.rs
+++ b/src/rust/engine/watch/src/lib.rs
@@ -53,7 +53,6 @@ struct Inner {
   watcher: RecommendedWatcher,
   executor: Executor,
   liveness: Receiver<String>,
-  enabled: bool,
   // Until the background task has started, contains the relevant inputs to launch it via
   // start_background_thread. The decoupling of creating the `InvalidationWatcher` and starting it
   // is to allow for testing of the background thread.
@@ -74,7 +73,6 @@ impl InvalidationWatcher {
     executor: Executor,
     build_root: PathBuf,
     ignorer: Arc<GitignoreStyleExcludes>,
-    enabled: bool,
   ) -> Result<Arc<InvalidationWatcher>, String> {
     // Inotify events contain canonical paths to the files being watched.
     // If the build_root contains a symlink the paths returned in notify events
@@ -87,28 +85,26 @@ impl InvalidationWatcher {
       .map_err(|e| format!("Failed to begin watching the filesystem: {}", e))?;
 
     let (liveness_sender, liveness_receiver) = crossbeam_channel::unbounded();
-    if enabled {
-      // On darwin the notify API is much more efficient if you watch the build root
-      // recursively, so we set up that watch here and then return early when watch() is
-      // called by nodes that are running. On Linux the notify crate handles adding paths to watch
-      // much more efficiently so we do that instead on Linux.
-      if cfg!(target_os = "macos") {
-        watcher
-          .watch(canonical_build_root.clone(), RecursiveMode::Recursive)
-          .map_err(|e| {
-            format!(
-              "Failed to begin recursively watching files in the build root: {}",
-              e
-            )
-          })?
-      }
+
+    // On darwin the notify API is much more efficient if you watch the build root
+    // recursively, so we set up that watch here and then return early when watch() is
+    // called by nodes that are running. On Linux the notify crate handles adding paths to watch
+    // much more efficiently so we do that instead on Linux.
+    if cfg!(target_os = "macos") {
+      watcher
+        .watch(canonical_build_root.clone(), RecursiveMode::Recursive)
+        .map_err(|e| {
+          format!(
+            "Failed to begin recursively watching files in the build root: {}",
+            e
+          )
+        })?
     }
 
     Ok(Arc::new(InvalidationWatcher(Mutex::new(Inner {
       watcher,
       executor,
       liveness: liveness_receiver,
-      enabled,
       background_task_inputs: Some((
         ignorer,
         canonical_build_root,
@@ -254,13 +250,14 @@ impl InvalidationWatcher {
   /// Add a path to the set of paths being watched by this invalidation watcher, non-recursively.
   ///
   pub async fn watch(self: &Arc<Self>, path: PathBuf) -> Result<(), notify::Error> {
+    if cfg!(target_os = "macos") {
+      // Short circuit here if we are on a Darwin platform because we should be watching
+      // the entire build root recursively already.
+      return Ok(());
+    }
+
     let executor = {
       let inner = self.0.lock();
-      if cfg!(target_os = "macos") || !inner.enabled {
-        // Short circuit here if we are on a Darwin platform because we should be watching
-        // the entire build root recursively already, or if we are not enabled.
-        return Ok(());
-      }
       inner.executor.clone()
     };
 

--- a/src/rust/engine/watch/src/tests.rs
+++ b/src/rust/engine/watch/src/tests.rs
@@ -33,7 +33,7 @@ async fn setup_watch(
   file_path: PathBuf,
 ) -> Arc<InvalidationWatcher> {
   let executor = Executor::new(Handle::current());
-  let watcher = InvalidationWatcher::new(executor, build_root, ignorer, /*enabled*/ true)
+  let watcher = InvalidationWatcher::new(executor, build_root, ignorer)
     .expect("Couldn't create InvalidationWatcher");
   watcher.watch(file_path).await.unwrap();
   watcher

--- a/tests/python/pants_test/init/test_options_initializer.py
+++ b/tests/python/pants_test/init/test_options_initializer.py
@@ -22,16 +22,9 @@ class OptionsInitializerTest(unittest.TestCase):
     def test_global_options_validation(self):
         # Specify an invalid combination of options.
         ob = OptionsBootstrapper.create(
-            args=[
-                "--backend-packages=[]",
-                "--backend-packages2=[]",
-                "--v2",
-                "--no-v1",
-                "--loop",
-                "--no-enable-pantsd",
-            ]
+            args=["--backend-packages=[]", "--backend-packages2=[]", "--remote-execution",]
         )
         build_config = BuildConfigInitializer.get(ob)
         with self.assertRaises(OptionsError) as exc:
             OptionsInitializer.create(ob, build_config)
-        self.assertIn("The `--loop` option requires `--enable-pantsd`", str(exc.exception))
+        self.assertIn("The `--remote-execution` option requires", str(exc.exception))

--- a/tests/python/pants_test/pantsd/test_watchman_launcher.py
+++ b/tests/python/pants_test/pantsd/test_watchman_launcher.py
@@ -10,7 +10,8 @@ from pants.testutil.test_base import TestBase
 
 class TestWatchmanLauncher(TestBase):
     def watchman_launcher(self, cli_options=()):
-        bootstrap_options = self.get_bootstrap_options(cli_options)
+        options = ("--watchman-enable", *cli_options)
+        bootstrap_options = self.get_bootstrap_options(options)
         return WatchmanLauncher.create(bootstrap_options)
 
     def create_mock_watchman(self, is_alive):

